exotica/src/KinematicTree.cpp

```diff
@@ -199,6 +199,33 @@ void KinematicTree::BuildTree(const KDL::Tree & RobotKinematics)
         if(Joint->IsControlled) ControlledJointsMap[Joint->Segment.getJoint().getName()] = Joint;
     }
 
+    // The ModelBaseType defined above refers to the base type of the overall
+    // robot model - not of the set of controlled joints. E.g. a floating-
+    // base robot can have a scene defined where the floating-base virtual
+    // joint is _not_ part of the planning group/scene. Thus we need to
+    // establish the BaseTye of the joint group, the ControlledBaseType:
+    //
+    // Generally assume fixed base, then assess semantic information
+    ControlledBaseType = BASE_TYPE::FIXED;
+    // If there is a virtual joint, we need to check in detail whether it is a
+    // planar or floating base
+    if (Model->getSRDF()->getVirtualJoints().size() > 0) {
+      std::string world_joint = Model->getSRDF()->getVirtualJoints()[0].name_;
+
+      // A floating base in KinematicTree always contains trans_x, trans_y,
+      // trans_z, rot_x, rot_y, rot_z (XYZRPY, 6-DoF)
+      if (std::find(ControlledJointsNames.begin(),
+                    ControlledJointsNames.end(),
+                    world_joint + "/trans_x") != ControlledJointsNames.end())
+        ControlledBaseType = BASE_TYPE::FLOATING;
+      // A planar base in Kinematica always contains x, y, theta (3-DoF)
+      else if (std::find(ControlledJointsNames.begin(),
+                         ControlledJointsNames.end(),
+                         world_joint + "/theta") !=
+               ControlledJointsNames.end())
+        ControlledBaseType = BASE_TYPE::PLANAR;
+    }
+
```

Comment:
Do you need to check SRDF here?
If the `ModelBaseType`  is Floating or Plannar, the `root_joint==world_joint`. So you can use ```RootJoint->getName()``` to get its name without using SRDF (line 212).
Then, if the root joint is part of the controlled joints ```ControlledBaseType=ModelBaseType``` (replacing lines 220-226).
You can then also check if the root joint is controlled when you add joints into the `ControlledJointsMap` vector at the end of ```BuildTree()```:
```
for(std::shared_ptr<KinematicElement> Joint : Tree)
    {
        Joint->ControlId = IsControlled(Joint);
        Joint->IsControlled = Joint->ControlId >= 0;
        if(Joint->IsControlled) ControlledJoints[Joint->ControlId] = Joint;
        TreeMap[Joint->Segment.getName()] = Joint;
        ModelJointsMap[Joint->Segment.getJoint().getName()] = Joint;
        if(Joint->IsControlled) 
        {
            ControlledJointsMap[Joint->Segment.getJoint().getName()] = Joint;
            \\ This is the new
            if(Joint->Segment.getJoint().getName()==RootJoint->getName() + "/trans_x") ControlledBaseType=BaseType;
        }
    }
```